/*
 * eframe-model - model模型数据模块
 * Copyright © 2020-2025 Linlan (open@linlan.net)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package net.linlan.sys.comm.service;

import java.util.List;
import java.util.Map;

import javax.annotation.Resource;

import org.springframework.stereotype.Service;

import com.github.pagehelper.Page;
import com.github.pagehelper.PageHelper;

import lombok.extern.slf4j.Slf4j;

import net.linlan.sys.comm.dao.CommLogDao;
import net.linlan.sys.comm.dto.CommLogDto;
import net.linlan.sys.comm.entity.CommLog;
import net.linlan.sys.comm.param.CommLogParam;

/**
 *
 * CommLog数据域:管理日志表服务类
 * @author Linlan
 * CreateTime 2018-03-02 17:26:29
 *
 */
@Service
@Slf4j
public class CommLogService {

    @Resource
    private CommLogDao dao;

    /** get the list of entity CommLog
     * 列表方法，返回列表的通用日志数据 {@link List} 对象，包含 {@link CommLog} 列表
     * @param map the input select conditions
     * @return 列表的通用日志数据 {@link List} 对象，包含 {@link CommLog} 列表
     */
    //@DataFilter(tableAlias = "B", owner = false)
    public List<CommLog> getList(Map<String, Object> map) {
        return dao.getList(map);
    }

    /** find the entity by input id, return entity
     * 对象详情方法，通过id查询对象{@link CommLog}
     * @param id the input id
     * @return {@link CommLog}
     */
    public CommLog findById(Long id) {
        return dao.findById(id);
    }

    /** save the entity with input object
     * 保存对象方法
     * @param commLog the input commLog
     */
    //@Transactional
    public void save(CommLog commLog) {
        commLog.init();
        dao.save(commLog);
    }

    /** save the entity with input list
     * 保存对象管理日志表，批量保存list
     * @param list {@link CommLog}
     */
    public void batchSave(List<CommLog> list) {
        dao.batchSave(list);
    }

    /** update the entity with input object
     * 更新对象方法
     * @param commLog the input commLog
     */
    //@Transactional
    public void update(CommLog commLog) {
        dao.update(commLog);
    }

    /** delete the entity by input id
     * 删除方法，通过id删除对象
     * @param id the input id
     */
    public void deleteById(Long id) {
        dao.deleteById(id);
    }

    /** batch delete the entity by input ids
     * 批量删除方法，通过ids删除对象
     * @param ids the input ids
     */
    //@Transactional
    public void deleteByIds(Long[] ids) {
        dao.deleteByIds(ids);
    }

    /** query the total count by input select conditions
     * 通过输入的条件查询记录总数
     * @param map the input select conditions
     * @return total count
     */
    public int getCount(Map<String, Object> map) {
        return dao.getCount(map);
    }

    /** get the page of dto CommLogDto
     * 列表分页方法，返回分页DTO对象列表
     * @param param the input param {@link CommLogParam }
     * @return 分页的通用日志数据 {@link Page} 对象，包含 {@link CommLog} 列表
     */
    //@DataFilter(tableAlias = "B", owner = false)
    public Page<CommLogDto> getPageDto(CommLogParam param) {
        PageHelper.startPage(param.getPage(), param.getLimit());
        return dao.getPageDto(param);
    }

    /** get the dto by input id, return dto
     * 对象详情方法，通过id查询对象{@link CommLogDto}
     * @param id the input id
     * @return {@link CommLogDto}
     */
    public CommLogDto getDtoById(Long id) {
        return dao.getDtoById(id);
    }

}
